src/alloc_pool.c

```diff
@@ -566,6 +566,7 @@ void qd_alloc_finalize(void)
         free_stack_chunks(&desc->global_pool->free_list);
         free(desc->global_pool);
         desc->global_pool = 0;
+        desc->header = 0;  // reset header, so we can initialize again later in qd_alloc (in subsequent test)
```

Comment:
This is actually something I think is missing in the dispatch code. Without it, it is not possible to stop the router and start it again, in the same process.